modules/konduitAnalyticsAdapter.js

```diff
@@ -0,0 +1,225 @@
+import { ajax } from '../src/ajax.js';
+import adapter from '../src/AnalyticsAdapter.js';
+import adapterManager from '../src/adapterManager.js';
+import * as utils from '../src/utils.js';
+import { targeting } from '../src/targeting.js';
+import { config } from '../src/config.js';
+import CONSTANTS from '../src/constants.json';
+
+const TRACKER_HOST = 'tracker.konduit.me';
+
+const analyticsType = 'endpoint';
+
+const eventDataComposerMap = {
+  [CONSTANTS.EVENTS.AUCTION_INIT]: obtainAuctionInfo,
+  [CONSTANTS.EVENTS.AUCTION_END]: obtainAuctionInfo,
+  [CONSTANTS.EVENTS.BID_REQUESTED]: obtainBidRequestsInfo,
+  [CONSTANTS.EVENTS.BID_TIMEOUT]: obtainBidTimeoutInfo,
+  [CONSTANTS.EVENTS.BID_RESPONSE]: obtainBidResponseInfo,
+  [CONSTANTS.EVENTS.BID_WON]: obtainWinnerBidInfo,
+  [CONSTANTS.EVENTS.NO_BID]: obtainNoBidInfo,
+};
+
+// This function is copy from prebid core
+function formatQS(query) {
+  return Object
+    .keys(query)
+    .map(k => Array.isArray(query[k])
+      ? query[k].map(v => `${k}[]=${v}`).join('&')
+      : `${k}=${query[k]}`)
+    .join('&');
+}
+
+// This function is copy from prebid core
+function buildUrl(obj) {
+  return (obj.protocol || 'http') + '://' +
+    (obj.host ||
+      obj.hostname + (obj.port ? `:${obj.port}` : '')) +
+    (obj.pathname || '') +
+    (obj.search ? `?${formatQS(obj.search || '')}` : '') +
+    (obj.hash ? `#${obj.hash}` : '');
+}
```

Comment:
Hi @jsnellbaker!
This is a good question. The main reason for the local copies is we would like the Konduit module to be compatible with previous Prebid versions.
Please let us know if there is a better way to maintain compatibility.

Reply:
The releases of Prebid are all snapshots of the repository taken at the respective times.  Your module would only be available for use with the version of Prebid.js that published it (and future versions).  It wouldn't inherently be backwards compatible, and I'm not sure there's any way within the current build/release process to make it work for older versions.

In regard to these functions, the local copies would seem to only help with the point I made before about potential worry of the core code updating and changing the logic slightly.  I wouldn't say it won't happen, but changes like that are considered very carefully in terms of the impact and we would take steps to ensure it's handled well for everyone.

Reply:
@jsnellbaker thank you for these details, everything makes sense.

Our use case is in fact to be backwards compatible.

We have a number of existing publishers that are not ready to upgrade their Prebid version yet. For us it would be easier to have a backwards compatible module that can be used with previous Prebid versions without code modification.

This is the main reason we have the local copies and hopefully it is acceptable to leave it as is.
Please let us know if you have any concerns.